packages/babel-parser/test/fixtures/experimental/pipeline-operator/proposal-smart-topic-style,-generator-yield/input.js

```diff
@@ -0,0 +1,3 @@
+function * f () {
+  return x |> (yield #);
```

Comment:
It does not work without parentheses, because `yield` is a very loose expression, looser than the pipe operator (and, indeed, looser than arrow functions or even assignment). The [pipe operator’s subexpressions are at the conditional-operator level of precedence](https://jschoi.org/18/es-smart-pipelines/spec#prod-PipeExpression). `x |> (yield #) |> # + 1` must have the parentheses just like how `1 + (yield #)` must have the parentheses.